src/ucs/datastruct/ptr_array.h

```diff
@@ -125,8 +126,10 @@ void *ucs_ptr_array_replace(ucs_ptr_array_t *ptr_array, unsigned index, void *ne
  * Iterate over all valid elements in the array.
  */
 #define ucs_ptr_array_for_each(_var, _index, _ptr_array) \
-    for (_index = 0; _index < (_ptr_array)->size; ++_index) \
-         if (!__ucs_ptr_array_is_free(_var = (void*)((_ptr_array)->start[_index]))) \
+    register unsigned elements = (_ptr_array)->n_occupied;\
```

Comment:
Using `register` is probably unnecessary

Reply:
Removed register.